clustered/client/src/main/java/org/ehcache/clustered/config/ClusteringServiceConfiguration.java

```diff
@@ -31,22 +38,67 @@
  * @author Clifford W. Johnson
  */
 // TODO: Should this accept/hold a *list* of URIs?
-// TODO: Add validation for connection URI(s)
-public class ClusteringServiceConfiguration
+// TODO: Add validation for cluster URI(s)
+public final class ClusteringServiceConfiguration
     implements ServiceCreationConfiguration<ClusteringService>,
     CacheManagerConfiguration<PersistentCacheManager> {
 
-  private final URI connectionUrl;
+  private final URI clusterUri;
+  private final String defaultServerResource;
+  private final Map<String, PoolDefinition> pools;
+
+  public ClusteringServiceConfiguration(final URI clusterUri, String defaultServerResource, Map<String, PoolDefinition> pools) {
+    if (clusterUri == null) {
+      throw new NullPointerException("clusterUri");
+    }
+    if (defaultServerResource == null) {
+      throw new NullPointerException("defaultServerResource");
+    }
+    this.clusterUri = clusterUri;
+    this.defaultServerResource = defaultServerResource;
+    this.pools = unmodifiableMap(new HashMap<String, PoolDefinition>(pools));
+  }
 
-  public ClusteringServiceConfiguration(final URI connectionUrl) {
-    if (connectionUrl == null) {
-      throw new NullPointerException("connectionUrl");
+  public ClusteringServiceConfiguration(URI clusterUri, Map<String, PoolDefinition> pools) {
+    if (clusterUri == null) {
+      throw new NullPointerException("clusterUri");
```

Comment:
Let's have the message a bit more explicit here ... "clusterUri cannot be null" or something like that